src/primitives.ts

```diff
@@ -955,76 +955,89 @@ export class PieChart {
       nextAngle -= sample[this.slicingVariable] * normedRatio;
-      pieParts.push(new PiePart(center[0], center[1], radius, initAngle, nextAngle, true));
+
+      let newPart = new PiePart(radius, initAngle, nextAngle);
+      newPart.color = 'hsl('+ colorRadius +', 50%, 50%, 90%)';
+      pieParts.push(newPart);
+
       initAngle = nextAngle;
     })
     return pieParts
   }
 }
 
-
 export class PiePart {
   hidden_color: string = '';
-  isMouseOver: boolean = false;
-  isSelected: boolean = false;
   path: Path2D;
   clicked: boolean = false;
   color: string = '';
+  readonly overfliedColor: string = string_to_hex('lightskyblue');
+  readonly selectedColor: string = string_to_hex("red");
+  readonly center: [number, number] = [0, 0];
 
   /**
-   * @param centerX the center x coordinate
-   * @param centerY the center y coordinate
+   * @param center the center coordinates [x, y]
    * @param radius radius
    * @param initAngle in radian
    * @param endAngle in radian
-   * @param anticlockwise true or false whether you want the arc the be drawn anticlockwise or not.
    */
   constructor(
-    public centerX: number,
-    public centerY: number,
     public radius: number,
     public initAngle: number, // Warning: canvas' angles are clockwise-oriented. For example: pi/2 rad is below -pi/2 rad.
-    public endAngle: number,
-    public anticlockwise: boolean = true,
-    public name: string = '') 
+    public endAngle: number) 
     {
       this.hidden_color = genColor();
       this.path = this.buildPath();
     }
 
   private buildPath(): Path2D {
     const path = new Path2D();
-    path.moveTo(this.centerX, this.centerY);
+    path.moveTo(this.center[0], this.center[1]);
     path.arc(
-      this.centerX,
-      this.centerY,
+      this.center[0],
+      this.center[1],
       this.radius,
       this.initAngle,
       this.endAngle,
-      this.anticlockwise);
+      true);
     return path
   }
 
   draw(context: CanvasRenderingContext2D, mvx: number, mvy: number, scale: number, X: number, Y: number): void {
     /**
-     * @param mvx x coordinate of the local frame in the canvas
-     * @param mvy y coordinate of the local frame in the canvas
+     * @param mvx x coordinate of the local frame in the canvas (center of the screen relatively to the canvas' high left corner)
+     * @param mvy y coordinate of the local frame in the canvas (center of the screen relatively to the canvas' high left corner)
      * @param scale ratio applied on data coordinates to manage mouse scrolling
-     * @param X x coordinate of the absolute frame (the canvas' one)
-     * @param Y y coordinate of the absolute frame (the canvas' one)
+     * @param X x coordinate of the offset of the drawing zone (0 if not in multiplot)
+     * @param Y y coordinate of the offset of the drawing zone (0 if not in multiplot)
      */
     const matrix = context.getTransform();
     context.transform(scale, 0, 0, scale, mvx + X, mvy + Y);
```

Comment:
This transform is done as many times as the number of parts. 
I think that we could do it only once in `drawPieChart` method